openday_scavenger/puzzles/xray_filters/static/index.html

```diff
@@ -0,0 +1,145 @@
+                            beam.style.visibility = 'visible';
+                            beam.style.opacity = '1';
+                            activeBeams.push(beam); // Collect visible beams
+                        } else {
+                            // Filter up
+                            filter.classList.remove(pair.filterClass.replace('.', '') + '-down');
+                            beam.style.visibility = 'hidden';
+                            beam.style.opacity = '0';
+                        }
+                    });
+
+                    // Apply progressive dimming in **reverse order**
+                    activeBeams.reverse().forEach((beam, index) => {
+                        // Clear previous dimming classes
+                        beam.classList.remove('dim-2', 'dim-3', 'dim-4');
+
+                        // Apply appropriate dimming class (reverse order)
+                        beam.classList.add(`dim-${index + 1}`);
+
+                        // forceRepaint(beam);
+                    });
+                }
+
+                // Loop through each pair and add event listeners to toggle 'down' class and update beams
+                checkboxFilterPairs.forEach(pair => {
+                    const checkbox = document.getElementById(pair.checkboxId);
+                    checkbox.addEventListener('change', updateBeamsAndFilters);
+                });
+
+                // Handle attenuation calculations for all checkboxes
+                function handleCheckboxClick() {
+                    let target_attenuation = 1;
+                    const checkboxes = document.querySelectorAll('.checkbox');
+
+                    // Calculate attenuation based on checked filters
+                    checkboxes.forEach(checkbox => {
+                        if (checkbox.checked) {
+                            target_attenuation *= parseFloat(checkbox.value);
+                        }
+                    });
+
+                    // Display the target attenuation
+                    document.getElementById('output').innerText = 'Target attenuation: ' + Math.round((1-target_attenuation)*100*10)/10 + '%';
+                    document.getElementById('answer').value = Math.round((1-target_attenuation)*100*10)/10;
+                }
+
+                // Add event listeners to all checkboxes for attenuation
+                const checkboxes = document.querySelectorAll('.checkbox');
+                checkboxes.forEach(checkbox => {
+                    checkbox.addEventListener('click', handleCheckboxClick);
+                });
+            });
+        </script>
+    </head>
+    <body>
+        <div class="title-banner">
+            <h1>X-ray Filter Puzzle</h1>
+        </div>
+        <div class="main-content">
+            <p>Sometimes, the x-rays are too strong(!). We need to attenuate them (make them weaker.) We do this by using filters to stop some of those pesky x-rays.</p>
```

Comment:
I prefer the word "bright" or "intense" instead of strong. And then "less bright" or "less intense" instead of weaker.
My preference is for X-ray to have a capital X, but that isn't critical.

Reply:
done!